References
=========================

Main References
----------------

The main referenced paper are:

- Shuni Wu, Yaqing Xu, Qingzhao Zhang, and Shuangge Ma. Gene-environment interaction analysis via deep learning. Genetic epidemiology, 47(3):261–286, 2023.

This paper is available at https://doi.org/10.1002/gepi.22518.

- Rui Ren, Kuangnan Fang, Qingzhao Zhang, and Shuangge Ma. Functansnp: an r package for functional analysis of dense snp data (with interactions). Bioinformatics, 39(12):btad741, 2023.

This paper is available at https://doi.org/10.1093/bioinformatics/btad741.


Complete References
--------------------

The complete referenced paper are listed below:

[1] Hugues Aschard, Martin D Tobin, Dana B Hancock, David Skurnik, Akshay Sood, Alan James, Albert Vernon Smith, Ani W Manichaikul, Archie Campbell, Bram P Prins, and others. Evidence for large-scale gene-by-smoking interaction effects on pulmonary function. International journal of epidemiology, 46(3):894–904, 2017.

[2] Mairead L Bermingham, Ricardo Pong-Wong, Athina Spiliopoulou, Caroline Hayward, Igor Rudan, Harry Campbell, Alan F Wright, James F Wilson, Felix Agakov, Pau Navarro, and others. Application of high-dimensional feature selection: evaluation for genomic prediction in man. Scientific reports, 5(1):10312, 2015.

[3] Avshalom Caspi and Terrie E Moffitt. Gene–environment interactions in psychiatry: joining forces with neuroscience. Nature Reviews Neuroscience, 7(7):583–590, 2006.

[4] Travers Ching, Xun Zhu, and Lana X Garmire. Cox-nnet: an artificial neural network method for prognosis prediction of high-throughput omics data. PLoS computational biology, 14(4):e1006076, 2018.

[5] Robert Clarke, Habtom W Ressom, Antai Wang, Jianhua Xuan, Minetta C Liu, Edmund A Gehan, and Yue Wang. The properties of high-dimensional data spaces: implications for exploring gene and protein expression data. Nature reviews cancer, 8(1):37–49, 2008.

[6] Astrid Dempfle, André Scherag, Rebecca Hein, Lars Beckmann, Jenny Chang-Claude, and Helmut Schäfer. Gene–environment interactions for complex traits: definitions, methodological requirements and challenges. European Journal of Human Genetics, 16(10):1164–1172, 2008.

[7] Jianqing Fan and Runze Li. Variable selection via nonconcave penalized likelihood and its oracle properties. Journal of the American statistical Association, 96(456):1348–1360, 2001.

[8] Ruzong Fan, Yifan Wang, James L Mills, Alexander F Wilson, Joan E Bailey-Wilson, and Momiao Xiong. Functional linear models for association analysis of quantitative traits. Genetic epidemiology, 37(7):726–742, 2013.

[9] Youssef Idaghdour, Jacklyn Quinlan, Jean-Philippe Goulet, Joanne Berghout, Elias Gbeha, Vanessa Bruat, Thibault De Malliard, Jean-Christophe Grenier, Selma Gomez, Philippe Gros, and others. Evidence for additive and interaction effects of host genotype and infection in malaria. Proceedings of the National Academy of Sciences, 109(42):16786–16793, 2012.

[10] Diego Jarquín, José Crossa, Xavier Lacaze, Philippe Du Cheyron, Joëlle Daucourt, Josiane Lorgeou, François Piraux, Laurent Guerreiro, Paulino Pérez, Mario Calus, and others. A reaction norm model for genomic selection using high-dimensional genomic and environmental data. Theoretical and applied genetics, 127:595–607, 2014.

[11] Matthew Kerin and Jonathan Marchini. A non-linear regression method for estimation of gene–environment heritability. Bioinformatics, 36(24):5632–5639, 12 2020.

[12] Henrik Källberg, Leonid Padyukov, Robert M Plenge, Johan Rönnelid, Peter K Gregersen, Annette HM van der Helm-van, Rene EM Toes, Tom W Huizinga, Lars Klareskog, Lars Alfredsson, and others. Gene-gene and gene-environment interactions involving hla-drb1, ptpn22, and smoking in two subsets of rheumatoid arthritis. The American Journal of Human Genetics, 80(5):867–875, 2007.

[13] Eric Lai. Application of snp technologies in medicine: lessons learned and future challenges. Genome Research, 11(6):927–929, 2001.

[14] Xinyi Lin, Seunggeun Lee, David C Christiani, and Xihong Lin. Test for interactions between a genetic marker set and environment in generalized linear models. Biostatistics, 14(4):667–681, 2013.

[15] Jin Liu, Jian Huang, Yawei Zhang, Qing Lan, Nathaniel Rothman, Tongzhang Zheng, and Shuangge Ma. Identification of gene-environment interactions in cancer studies using penalization. Genomics, 102(4):189–194, 2013.

[16] Stephen B Manuck and Jeanne M McCaffery. Gene-environment interaction. Annual review of psychology, 65(1):41–70, 2014.

[17] Kimberly McAllister, Leah E Mechanic, Christopher Amos, Hugues Aschard, Ian A Blair, Nilanjan Chatterjee, David Conti, W James Gauderman, Li Hsu, Carolyn M Hutter, and others. Current challenges and new opportunities for gene-environment interaction studies of complex diseases. American journal of epidemiology, 186(7):753–761, 2017.

[18] Marcus R Munafò, Caroline Durrant, Glyn Lewis, and Jonathan Flint. Gene× environment interactions at the serotonin transporter locus. Biological psychiatry, 65(3):211–219, 2009.

[19] Rui Ren, Kuangnan Fang, Qingzhao Zhang, and Shuangge Ma. Functansnp: an r package for functional analysis of dense snp data (with interactions). Bioinformatics, 39(12):btad741, 2023.

[20] JE Seeb, Gary Carvalho, Lorenz Hauser, Kerry Naish, Steven Roberts, and LW Seeb. Single-nucleotide polymorphism (snp) discovery and applications of snp genotyping in nonmodel organisms. Molecular ecology resources, 2011.

[21] Noha Sharafeldin, Martha L Slattery, Qi Liu, Conrado Franco-Villalobos, Bette J Caan, John D Potter, and Yutaka Yasui. A candidate-pathway approach to identify gene-environment interactions: analyses of colon cancer risk and survival. Journal of the National Cancer Institute, 107(9):djv160, 2015.

[22] Jooyong Shim, Changha Hwang, Sunjoo Jeong, and Insuk Sohn. Semivarying coefficient least-squares support vector regression for analyzing high-dimensional gene-environmental data. Journal of applied statistics, 45:1370–1381, 2018.

[23] D Leland Taylor, David A Knowles, Laura J Scott, Andrea H Ramirez, Francesco Paolo Casale, Brooke N Wolford, Li Guan, Arushi Varshney, Ricardo D’Oliveira Albanus, Stephen CJ Parker, and others. Interactions between genetic variation and cellular environment in skeletal muscle gene expression. PLoS One, 13(4):e0195788, 2018.

[24] Barinder Thind, Kevin Multani, and Jiguo Cao. Deep learning with functional inputs. Journal of Computational and Graphical Statistics, 32(1):171–180, 2023.

[25] Duncan Thomas. Gene–environment-wide association studies: emerging approaches. Nature Reviews Genetics, 11:259–272, 2010.

[26] Duncan Thomas. Methods for investigating gene-environment interactions in candidate pathway and genome-wide association studies. Annual review of public health, 31(1):21–36, 2010.

[27] Samuel J Virolainen, Andrew VonHandorf, Kenyatta CMF Viel, Matthew T Weirauch, and Leah C Kottyan. Gene–environment interactions and their impact on human health. Genes & Immunity, 24(1):1–11, 2023.

[28] Jane-Ling Wang, Jeng-Min Chiou, and Hans-Georg Müller. Review of functional data analysis. Annual Review of Statistics and its application, 3(1):257–295, 2016.

[29] Mengyun Wu and Shuangge Ma. Robust genetic interaction analysis. Briefings in bioinformatics, 20(2):624–637, 2019.

[30] Mengyun Wu, Qingzhao Zhang, and Shuangge Ma. Structured gene-environment interaction analysis. Biometrics, 76(1):23–35, 2020.

[31] Shuni Wu, Yaqing Xu, Qingzhao Zhang, and Shuangge Ma. Gene–environment interaction analysis via deep learning. Genetic epidemiology, 47(3):261–286, 2023.

[32] Cun-Hui Zhang. Nearly unbiased variable selection under minimax concave penalty. The Annals of Statistics, 38(2):894–942, 2010.

[33] Guolin Zhao, Rachel Marceau, Daowen Zhang, and Jung-Ying Tzeng. Assessing gene-environment interactions for common and rare variants with binary traits using gene-trait similarity regression. Genetics, 199(3):695–710, 2015.
